File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    rank = {6:1,
           5:2,
           4:3,
           3:4,
           2:5,
           1:6,
           0:6}
    lot = 0
    zero_num = 0
    for win in win_nums:
        for lotto in lottos:
            if win == lotto:
                lot += 1
    for zero in lottos:
        if zero == 0:
            zero_num += 1
    answer.append(rank[lot+zero_num])
    answer.append(rank[lot])
    return answer

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
import java.util.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] rank = {6,6,5,4,3,2,1};
        int[] answer = new int[2];

        for(int lotto : lottos) {
            if(lotto == 0) {
                answer[0]++;
                continue;
            }
            for(int win : win_nums) {
                if(win == lotto) {
                    answer[0]++;
                    answer[1]++;
                }
            }
        }
        answer[0] = rank[answer[0]];
        answer[1] = rank[answer[1]];
        return answer;
    }
}